src/test/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforceStreamingSourceConfigTest.java

```diff
@@ -0,0 +1,159 @@
+        QueryResult queryResult = mock(QueryResult.class);
+        PartnerConnection partnerConnection = mock(PartnerConnection.class);
+        Mockito.when(partnerConnection.query(anyString())).thenReturn(queryResult);
+        Mockito.when(queryResult.getRecords()).thenReturn(sObjectArray);
+        assertEquals(null, SalesforceStreamingSourceConfig.fetchPushTopicByName(partnerConnection, "test"));
+    }
+
+    @Test
+    public void fetchPushTopicByNameTestWhenSingleRecord() throws ConnectionException {
+        SObject sObject = mock(SObject.class);
+        SObject[] sObjectArray = {sObject};
+        QueryResult queryResult = mock(QueryResult.class);
+        PartnerConnection partnerConnection = mock(PartnerConnection.class);
+        Mockito.when(partnerConnection.query(anyString())).thenReturn(queryResult);
+        Mockito.when(queryResult.getRecords()).thenReturn(sObjectArray);
+        assertEquals(sObject, SalesforceStreamingSourceConfig.fetchPushTopicByName(partnerConnection, "test"));
+    }
+
+    @Test
+    public void fetchPushTopicByNameTestWhenMultipleRecord() throws ConnectionException {
+        SObject sObject = mock(SObject.class);
+        SObject[] sObjectArray = {sObject, sObject};
+        QueryResult queryResult = mock(QueryResult.class);
+        PartnerConnection partnerConnection = mock(PartnerConnection.class);
+        Mockito.when(partnerConnection.query(anyString())).thenReturn(queryResult);
+        Mockito.when(queryResult.getRecords()).thenReturn(sObjectArray);
+        try {
+            SalesforceStreamingSourceConfig.fetchPushTopicByName(partnerConnection, "test");
+        } catch (IllegalStateException e) {
+            assertEquals("Excepted one or zero pushTopics with name = 'test' found 2", e.getMessage());
+        }
+    }
+
+    @Test
+    public void getSObjectQueryTest() throws Exception {
+        mockStatic(SObjectDescriptor.class);
+        mockStatic(SalesforceQueryUtil.class);
+        mockStatic(SObjectFilterDescriptor.class);
+        SObjectDescriptor sObjectDescriptor = spy(new SObjectDescriptor("test", new ArrayList<>()));
+        SObjectFilterDescriptor sObjectFilterDescriptor = mock(SObjectFilterDescriptor.class);
+        SalesforceStreamingSourceConfig config = mock(SalesforceStreamingSourceConfig.class);
+        PowerMockito.when(config.canAttemptToEstablishConnection()).thenReturn(true);
+        PowerMockito.when(sObjectDescriptor.getFieldsNames()).thenReturn(new ArrayList<>());
+        PowerMockito.when(SObjectFilterDescriptor.noOp()).thenReturn(sObjectFilterDescriptor);
+        PowerMockito.when(SObjectDescriptor.fromName(anyString(), any(), any())).thenReturn(sObjectDescriptor);
+        PowerMockito.when(SalesforceQueryUtil.createSObjectQuery(any(), anyString(), any())).thenReturn("sObjectQuery");
+        try {
+            Whitebox.invokeMethod(config, "getSObjectQuery");
+        } catch (Exception e) {
+            e.getMessage();
```

Comment:
fail if exception is not thrown

Reply:
Failure Assertion added, if no exception thrown